### DISCUSSION TITLE
Am I "cheating" if I use Visual Studio then copying my answer to Codewars?

### DISCUSSION THREAD

**Discussion by dannyfigs33** [codewars/codewars.com#2658] [Q&A] [answered]
Is it wrong if I'm using the features of the IDE to help me code? I find that a lot of the katas are pretty easy for me to find a solution to while using visual studio (with the intellisense options to scroll through, displaying method definitions & properties), but in the codewars browser IDE, it takes me a while & a lot of googling to try to solve an answer, since it doesn't help you out at all. 

Is it fine as a practice to use visual studio IDE to help me code, and then copying and pasting my solution to codewars?

**Comment by Blind4Basics** [comment, marked as answer]
absolutely no problem with that.

**Comment by Stefan3002** [comment]
I guess you are referring to the autocomplete function of VS Code and possibly other IDEs? Imo, no, that is fine, it is normal to forget the exact name of some things, you just have to know where to search for it. Cheers!

**Comment by nacimux** [comment]
I personally use vsCode or codesandbox to reason about code challenge, it helps me focus on resolving the challenge in my "comfort zone". The only thing that i do not use is obviously Github Copilot.
